### PR TITLE
Topic/level indicator

### PR DESCRIPTION
**Pull request by scztt**
This re-enables led-mode for level indicators. 

It re-works the style numbering slightly. Style #2, led indicators, becomes style #1. Style #1 used to be black bars - In lieu of this mode, I added parameters to change the colors of the level meters - so, you can achieve the same result by using style = 1 plus meterColor = Color.black. This API change seems acceptable considering legacy code that used the old styles would already have been broken for several years.

LevelIndicator:stepWidth becomes the primary way of setting the size of the led rectangles. The old numSteps is supported as well, but in 95% of cases the width of the step is a more useful value to set than the number of steps.

Finally, there's some refactoring of the way the level indicators are drawn - mainly, using transforms to get to a more reasonable coordinate system.

### CODE REVIEW

File: QtCollider/widgets/QcLevelIndicator.h [hunk 1]
```diff
@@ -61,6 +77,8 @@ private Q_SLOTS:
   bool _drawPeak;
   int _ticks;
   int _majorTicks;
+	float _stepWidth;
+	int _style;
```
Comment (timblechmann) [inline QtCollider/widgets/QcLevelIndicator.h]: is it possible to convert _style from int to an enum? that will make the code a bit more readable/expressive.

File: QtCollider/widgets/QcLevelIndicator.h [hunk 2]
```diff
@@ -28,6 +28,8 @@
 #include <QWidget>
 #include <QTimer>
 
+static const double PI = 3.14159265358979323846264338327950288419717;
```
Comment (timblechmann) [inline QtCollider/widgets/QcLevelIndicator.h]: M_PI? or pi from SC_Constants.h?

File: QtCollider/widgets/QcLevelIndicator.h (hunk 1 above)
Comment (scztt) [inline QtCollider/widgets/QcLevelIndicator.h]: Good call, I'll push that change, along with a symbol<>enum conversion on the SC side, following the pattern of other QT stuff.

File: QtCollider/widgets/QcLevelIndicator.h (hunk 2 above)
Comment (scztt) [inline QtCollider/widgets/QcLevelIndicator.h]: Cruft, will remove.

**Pull request closed without merging**